Source/deviceinfo/DeviceInfo.cpp

```diff
@@ -611,106 +614,574 @@ class DeviceInfoLink : public WPEFramework::RPC::SmartInterfaceType<WPEFramework
+        {
+            strncpy(buffer, _platformName.Value().c_str(), _platformName.Value().length());
+            *length = _platformName.Value().length();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_platform_name(buffer,length);
+            if(result == Core::ERROR_NONE)
+            {
+                _platformName = buffer;
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_maximum_output_resolution(deviceinfo_output_resolution_t* value)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_maxOutputResolution.IsSet())
+        {
+            *value = _maxOutputResolution;
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_maximum_output_resolution(value);
+            if(result == Core::ERROR_NONE)
+            {
+                _maxOutputResolution = *value;
+            }
+        }
+        return result;
+    }
+
+    uint32_t Deviceinfo_output_resolutions(deviceinfo_output_resolution_t value[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_outputResolution.size() > 0)
+        {
+            std::copy(_outputResolution.begin(), _outputResolution.end(), value);
+            *length = _outputResolution.size();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_output_resolutions(value, length);
+            if(result == Core::ERROR_NONE)
+            {
+                _outputResolution.insert(_outputResolution.begin(), value, value + (*length));
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_video_outputs(deviceinfo_video_output_t value[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_videoOutput.size() > 0)
+        {
+            std::copy(_videoOutput.begin(), _videoOutput.end(), value);
+            *length = _videoOutput.size();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_video_outputs(value, length);
+            if(result == Core::ERROR_NONE)
+            {
+                _videoOutput.insert(_videoOutput.begin(), value, value + (*length));
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_audio_outputs(deviceinfo_audio_output_t value[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_audioOutput.size() > 0)
+        {
+            std::copy(_audioOutput.begin(), _audioOutput.end(), value);
+            *length = _audioOutput.size();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_audio_outputs(value, length);
+            if(result == Core::ERROR_NONE)
+            {
+                _audioOutput.insert( _audioOutput.begin(), value, value + (*length));
+            }
+        }
+        return result;
+    }
+
+private:
+
+    void setHdrSupport(const bool& hdr)
+    {
+        _hdr_atmos_cec |= (1 & hdr) << 7;
```

Comment:
Not very important but what is the use of the "1 &"? Think it also works the same without it